### ISSUE TITLE
`OwnsOne` does not convert correctly. 

### ISSUE THREAD

**Issue by jamalabo1** [efcore/EFCore.NamingConventions#17]
the naming conventions do not convert the naming correctly,
instead of `location_floor`, it converts to `floor`

the version I'm using: `1.1.0`

```
  public class Room
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public RoomLocation Location { get; set; }

        public DateTime CreationDate { get; set; }
        public DateTime ModificationDate { get; set; }
        
    }
    public class RoomLocation
    {
        public int Floor { get; set; }
        public int Building { get; set; }
        public int Room { get; set; }
    }
```
```
modelBuilder.Entity<Room>(p => { p.OwnsOne(builder => builder.Location); });
```

#### with `UseSnakeCaseNamingConvention()`: 

```(id, creation_date, modification_date, title, building, floor, room)```
#### without `UseSnakeCaseNamingConvention()`:

```("Id", "CreationDate", "ModificationDate", "Title", "Location_Building", "Location_Floor", "Location_Room")```

**Comment by jamalabo1** [comment]
@roji any updates?

**Comment by roji** [comment]
Sorry that I haven't responded, these are very busy times... I'll do my best to get around to this.

**Comment by jamalabo1** [comment]
@roji no worries, 
I'd also like to note that it does not track updated properties, even when I specify the name using fluent API

using the way below allows me to insert entities, but I can't update them because EFcore does not track them and set's the value to null instead.
```
            modelBuilder.Entity<Room>().OwnsOne(p => p.Location,
                cb =>
                {
                    cb.Property(x => x.Building).HasColumnName("location_building");
                    cb.Property(x => x.Floor).HasColumnName("location_floor");
                    cb.Property(x => x.Room).HasColumnName("location_room");
                });
```

**Comment by roji** [comment]
Yeah, owned entities are a bit tricky because names get changed under the hood. I'll try to look into it as soon as I can.

**Comment by flashQarl** [comment]
Hi @jamalabo1 !

You need to specify the name of the column that has the foreign key in database.

Try something like this:
`modelBuilder.Entity<Room>().OwnsOne(p => p.Location,
    cb =>
    {
        cb.ToTable("room_locations"); // Only if you have table splitting

        cb.WithOwner().HasForeignKey("location_room"); // Foreign key with database column name

        cb.Property(x => x.Building).HasColumnName("location_building");
        cb.Property(x => x.Floor).HasColumnName("location_floor");
        cb.Property(x => x.Room).HasColumnName("location_room");
    });`

**Comment by jamalabo1** [comment]
@flashQarl I'm not interested in splitting tables
But I have done this workaround

```
    public class Room : DbEntity
    {
        public string Title { get; set; }

        [NotMapped]
        public RoomLocation Location
        {
            get =>
                new RoomLocation
                {
                    Building = LocationBuilding,
                    Floor = LocationFloor,
                    Room = LocationRoom
                };
            set
            {
                LocationBuilding = value.Building;
                LocationFloor = value.Floor;
                LocationRoom = value.Room;
            }
        }

        [JsonIgnore] public int LocationFloor { get; set; }

        [JsonIgnore] public int LocationBuilding { get; set; }

        [JsonIgnore] public int LocationRoom { get; set; }

    }
```

**Comment by roji** [comment]
Fixed in 5.0.0-preview5
